packages/replay-next/src/contexts/ExpandablesContext.tsx

```diff
@@ -0,0 +1,24 @@
+import { PropsWithChildren, createContext, useCallback, useMemo } from "react";
+
+export type InspectorContextType = {
+  isExpanded(key: string): boolean | undefined;
+  setIsExpanded(key: string, expanded: boolean): void;
+};
+
+export const ExpandablesContext = createContext<InspectorContextType>({
+  isExpanded: () => undefined,
+  setIsExpanded: () => {},
+});
+
+export function ExpandablesContextRoot({ children }: PropsWithChildren) {
+  const expandedKeys = useMemo(() => new Map<string, boolean>(), []);
+  const isExpanded = useCallback((key: string) => expandedKeys.get(key), [expandedKeys]);
+  const setIsExpanded = useCallback(
+    (key: string, expanded: boolean) => expandedKeys.set(key, expanded),
+    [expandedKeys]
+  );
```

Comment:
I think this works but I wonder if you've considered the fact that `setIsExpanded` _sounds_ like it should trigger a state update (like other React "setters") when it won't actually in this case.

Maybe it would be nicer to name thing something more passive sounding– like "onExpandedChange" or something.

Alternately you could refactor `expandedKeys` to be stored in state but that's probably overkill.

Reply:
To me `setIsExpanded` sounds like it sets a value and I don't feel that ties it to React state updates. I don't like `onExpandedChange` because it doesn't say anything about what it does, only when it is expected to be called.

> Alternately you could refactor expandedKeys to be stored in state but that's probably overkill.

Yeah, I considered that but also saw it as overkill.

Reply:
> To me `setIsExpanded` sounds like it sets a value and I don't feel that ties it to React state updates. I don't like `onExpandedChange` because it doesn't say anything about what it does, only when it is expected to be called.

I'm going to push back pretty strongly on this. "value" and "setValue" have _very specific implications_ in React components. They suggest values stored in React state, that updating will trigger re-renders. I don't think we should use that naming convention for this.

Reply:
OK, I'll change it tomorrow (perhaps to `persistIsExpanded`?)

Reply:
I like that name!